### ISSUE TITLE
Panning is too difficult

### ISSUE THREAD

**Issue by f3rno** [gloomyson/StarCraft#4]
The screen does not pan/scroll if the mouse passes the edge of the browser window.

**Comment by gloomyson** [comment]
You mean scroll cannot work? But it works in my side, I set Map.triggerMargin as 20px, which means when mouse hover less than 20px from edge should scroll

**Comment by f3rno** [comment]
Yep, but if you scroll past the edge, it stops. It should continue scrolling in the direction the mouse pointer was last seen traveling in. I have my tabs on the left side of the screen in Firefox, so I have a very hard time scrolling left (I have to keep my cursor right on the inside tab edge).

**Comment by Lewiscowles1986** [comment]
@f3rno this can be done, but it sounds like it will lead to more complaints and bugs, as what happens if I leave on right and come back on left or bottom or top? it would trigger scroll / pan, so it's undesirable. It falls under fixing a user being stupid, poor page design etc. 

Perhaps a game could enforce fullscreen as a trade-off, to prevent this happening as I would assume it would affect iframes and certain poor designs also. Maybe adding an event with callback would enable @f3rno to add their ideas, whilst keeping the core free of implementation specific details?

**Comment by f3rno** [comment]
@Lewiscowles1986 if the mouse re-enters the view from any direction, panning should adjust accordingly. If you leave the view on the right, it pans right; if you re-enter from the top, it pans upwards. This seems most intuitive, and it's how SC2 behaves in windowed mode (iirc).

Basically, if the mouse is detected in the panning region at the edge of the screen, start panning. If the mouse leaves the view, but does not leave the panning region first (i.e. goes further out and over the border), keep on panning.

**Comment by tvararu** [comment]
There is an HTML5 API that solves this problem neatly: http://caniuse.com/#feat=pointerlock

The best solution would be to use it where it's available, and fall back to mouse position polling (that will fail when users move the mouse too fast) where it's not.

**Comment by Lewiscowles1986** [comment]
@f3rno I'll leave it to @gloomyson choice, but personally I wouldn't want what I consider a well-intentioned poor interaction in something I built; and while I can see @tvararu point, at 25% browser usage I think such polyfills should be left to application specific implementations rather than the core framework (which is what I hope this is, as otherwise it's blatant copyright infringement)
